File: backend/src/MyDatingProfileHelper.Backend.Api/DTOs/SaveUserProfileRequest.cs
csharp
namespace MyDatingProfileHelper.Backend.Api.DTOs
{
    /// <summary>
    /// DTO для запроса на сохранение или обновление профиля пользователя через API.
    /// Используется для приема данных от клиента.
    /// </summary>
    public class SaveUserProfileRequest
    {
        /// <summary>
        /// Имя пользователя.
        /// </summary>
        public string? Name { get; set; }

        /// <summary>
        /// Список ссылок на фотографии пользователя.
        /// </summary>
        public List<string>? Photos { get; set; }

        /// <summary>
        /// Список интересов пользователя.
        /// </summary>
        public List<string>? Interests { get; set; }

        /// <summary>
        /// Список целей знакомства пользователя.
        /// </summary>
        public List<string>? DatingGoals { get; set; }

        /// <summary>
        /// Описание профиля пользователя.
        /// </summary>
        public string? Description { get; set; }

        /// <summary>
        /// Пол пользователя.
        /// </summary>
        public string? Gender { get; set; }
    }
}

File: app/src/main/java/com/jounik_projects/mydatingprofilehelper/data/network/model/SaveUserProfileRequest.kt
package com.jounik_projects.mydatingprofilehelper.data.network.model

/**
 * Класс данных для запроса сохранения профиля пользователя на бэкенде.
 * Соответствует SaveUserProfileRequest DTO на бэкенде.
 */
data class SaveUserProfileRequest(
    // TODO: Добавить свойства, соответствующие SaveUserProfileRequest на бэкенде
    // Например:
    // val name: String,
    // val photos: List<String>,
    // val interests: List<String>,
    // val datingGoals: List<String>,
    // val description: String,
    // val likes: List<String>, // Что нравится
    // val dislikes: List<String>, // Что не нравится
    // val badHabitsAttitude: String, // Отношение к вредным привычкам
// val height: Int?, // Рост
// val hairColor: String?, // Цвет волос
// val bodyType: String? // Телосложение
 val gender: String // Пол пользователя
)